src/StrategyConvexStaker.sol

```diff
@@ -0,0 +1,222 @@
+     * to deposit in the yield source.
+     */
+    function _deployFunds(uint256 _amount) internal override {
+        // the final true argument means we deposit + stake at the same time
+        booster.deposit(pid, _amount, true);
+    }
+
+    /**
+     * @dev Will attempt to free the '_amount' of 'asset'.
+     *
+     * The amount of 'asset' that is already loose has already
+     * been accounted for.
+     *
+     * This function is called during {withdraw} and {redeem} calls.
+     * Meaning that unless a whitelist is implemented it will be
+     * entirely permissionless and thus can be sandwiched or otherwise
+     * manipulated.
+     *
+     * Should not rely on asset.balanceOf(address(this)) calls other than
+     * for diff accounting purposes.
+     *
+     * Any difference between `_amount` and what is actually freed will be
+     * counted as a loss and passed on to the withdrawer. This means
+     * care should be taken in times of illiquidity. It may be better to revert
+     * if withdraws are simply illiquid so not to realize incorrect losses.
+     *
+     * @param _amount, The amount of 'asset' to be freed.
+     */
+    function _freeFunds(uint256 _amount) internal override {
+        rewardsContract.withdrawAndUnwrap(_amount, false);
+    }
+
+    /**
+     * @dev Internal function to harvest all rewards, redeploy any idle
+     * funds and return an accurate accounting of all funds currently
+     * held by the Strategy.
+     *
+     * This should do any needed harvesting, rewards selling, accrual,
+     * redepositing etc. to get the most accurate view of current assets.
+     *
+     * NOTE: All applicable assets including loose assets should be
+     * accounted for in this function.
+     *
+     * Care should be taken when relying on oracles or swap values rather
+     * than actual amounts as all Strategy profit/loss accounting will
+     * be done based on this returned value.
+     *
+     * This can still be called post a shutdown, a strategist can check
+     * `TokenizedStrategy.isShutdown()` to decide if funds should be
+     * redeployed or simply realize any profits/losses.
+     *
+     * @return _totalAssets A trusted and accurate account for the total
+     * amount of 'asset' the strategy currently holds including idle funds.
+     */
+    function _harvestAndReport()
+        internal
+        override
+        returns (uint256 _totalAssets)
+    {
+        uint256 _looseAssets = balanceOfWant();
+        if (_looseAssets > 0) {
+            _deployFunds(_looseAssets);
+        }
+
+        // claim any pending rewards
+        _claimRewards();
+
+        _totalAssets = balanceOfStake();
+    }
+
+    /**
+     * @notice Use to manually claim rewards from our staking contract.
+     * @dev Can only be called by management. Mostly helpful to make life easier for trade factory.
+     */
+    function manualRewardsClaim() external onlyKeepers {
```

Comment:
is there any pre-existing yearn infra we can activate here for keepers or we have to roll out our own implementation?